src/OrbitVulkanLayer/SubmissionTracker.h

```diff
@@ -976,6 +893,140 @@ class SubmissionTracker : public VulkanLayerProducer::CaptureStatusListener {
     command_buffer_to_state_.erase(command_buffer);
   }
 
+  void PersistSingleCommandBufferOnSubmitUnsafe(
```

Comment:
Why "unsafe"? Because this doesn't lock the mutex? Unless I'm missing something, this should never be called without actually holding the mutex. I would remove the "unsafe" in the name and add `mutex_.AssertHeld();` at the beginning.

Reply:
Yes that was regarding the Mutex. I used this pattern kind of across the whole layer. We used this in my previous project a lot to make it explicit to the caller of a method that is has some very specific assumptions, which I kind of liked. 
Anyways, the `AssertHeld` makes this assumption even more explicit, so I added it. For now, I removed the "unsafe" part of the name. But we are already in a mixture of this in the layer code base. I will make my mind about this, and address this in a separate PR (if needed).

Reply:
IMO, with `AssertHeld`, the "unsafe" is redundant or even wrong. You can't call it in an unsafe way anyway, so there's no need to know about it in the name. If you use it wrong, you will learn about it. 

An "unsafe" version of a method without `AssertHeld` suggests that there are use cases where I can call the function without holding the mutex. So unless these use cases exist, I would always use the assert and remove the "unsafe".

Reply:
Yep, with the concrete examples and the asserts, I agree.